### ISSUE TITLE
question about nn.Linear

### ISSUE THREAD

**Issue by harrylin-hyl** [lucastabelini/LaneATT#28]
"batch_anchor_features = batch_anchor_features.reshape(-1, self.anchor_feat_channels * self.fmap_h)"
I notice the feature is resized to (8000, 704) before the nn.Linear. But I do not understand the meaning of this, why not use (8, 1000, 704) to the linear layer. As far as I know,  the first dimension is for batch size, which is independent. 
plz answer my confusion, best wish to you.
Thanks!

**Comment by lucastabelini** [comment]
The features are independent of each other, so this tensor can be seen as a batch of 8000 proposals, where each proposal has 704 features (i.e., it is a batch of 8000 proposals, not a batch of 8 sets of features from each image).

**Comment by harrylin-hyl** [comment]
Thanks for your quick reply! Another question that confused me is why always getting different results in the same experiment setting? For example, I run Culane dataset on ResNet-34 but just get 76.25 F1 instead of 76.77 F1 in the paper. I run the "Culane_resnet34"  many times, but I can not get the same high performance.

**Comment by harrylin-hyl** [comment]
sorry, 76.68 F1 in the paper

**Comment by lucastabelini** [comment]
As written in the README.MD, the code is using my own implementation (Python) of CULane`s metric, but the metric reported in the paper uses CULane's official code (C++). Moreover, if you use PyTorch version other than 1.6 (the one used to train the models) the results are going to be a little bit different (also written in the README.MD). Therefore, make sure you are using PyTorch 1.6 with CUDA 10.2 and then use CULane's official metric implementation (the predictions are stored after running the inference with LaneATT).

**Comment by harrylin-hyl** [comment]
Thanks! I understand.
